packages/babel-plugin-transform-react-jsx-source/src/index.js

```diff
@@ -0,0 +1,56 @@
+
+ /**
+ * This adds {fileName, lineNumber} annotations to React component definitions
+ * and to jsx tag literals.
+ *
+ *
+ * == JSX Literals ==
+ *
+ * <sometag />
+ *
+ * becomes:
+ *
+ * <sometag __source={{fileName: 'this/file.js', lineNumber: 10}}/>
```

Comment:
This is more aligned with what ReactNative does.  Also, it's easier for people to pull out data if we wanted to parse/render it differently in the future (you can easily imagine a devtool doing that, rendering errors inline with the code, and you don't want them to have to parse the value).  Also, it's dev-mode anyway, so the extra allocations aren't a huge deal from a performance perspective.  I'm generally in favor of keeping data in a data structure so we don't just end up parsing a string later, when possible/practical.